File: scripts/git-push.py
import subprocess
import os

os.chdir('/vercel/share/v0-project')

try:
    print('[v0] Adding all changes...')
    subprocess.run(['git', 'add', '-A'], check=True)
    
    print('[v0] Checking git status...')
    result = subprocess.run(['git', 'status', '--porcelain'], capture_output=True, text=True)
    print('[v0] Changes:', result.stdout)
    
    print('[v0] Committing changes...')
    try:
        subprocess.run(['git', 'commit', '-m', 'fix: resolve CSS and build errors - fix leading-tight class'], check=True)
    except subprocess.CalledProcessError:
        print('[v0] Nothing to commit or commit skipped')
    
    print('[v0] Getting current branch...')
    result = subprocess.run(['git', 'rev-parse', '--abbrev-ref', 'HEAD'], capture_output=True, text=True, check=True)
    branch = result.stdout.strip()
    print(f'[v0] Current branch: {branch}')
    
    print('[v0] Pushing to GitHub...')
    subprocess.run(['git', 'push', 'origin', branch], check=True)
    
    print('[v0] ✅ Code pushed to GitHub successfully!')

except subprocess.CalledProcessError as e:
    print(f'[v0] Error: {e}')
    exit(1)
except Exception as e:
    print(f'[v0] Error: {e}')
    exit(1)

File: scripts/git-push.js
import { execSync } from 'child_process';
import path from 'path';
import { fileURLToPath } from 'url';

const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);
const projectRoot = path.join(__dirname, '..');

try {
  console.log('[v0] Configuring git...');
  execSync('git config user.email "v0@vercel.com"', { cwd: projectRoot, stdio: 'pipe' });
  execSync('git config user.name "v0 Bot"', { cwd: projectRoot, stdio: 'pipe' });

  console.log('[v0] Adding changes...');
  execSync('git add .', { cwd: projectRoot });

  console.log('[v0] Committing changes...');
  try {
    execSync('git commit -m "fix: resolve CSS and build errors - fix leading-tight class and complete component structure"', { cwd: projectRoot });
  } catch (e) {
    console.log('[v0] Nothing to commit or commit failed - continuing');
  }

  console.log('[v0] Pushing to GitHub...');
  execSync('git push origin HEAD:v0/rubabbhatti310504-6504-40469324', { cwd: projectRoot, stdio: 'inherit' });

  console.log('[v0] ✅ Code pushed to GitHub successfully');
  process.exit(0);
} catch (error) {
  console.error('[v0] Error:', error.message);
  process.exit(1);
}
